### PR TITLE
3-ary handler for resource

### PR DESCRIPTION
**Pull request by kalekale**

### CODE REVIEW

**Comment by miikka** [comment]
I'd recommend squashing commits like `moved try block` and `fixed indent` into the main commit. They're basically just changing whitespace, so they don't have much historical value. You can do it with `git rebase -i`.

It'd be great if you can figure out how to write an unit test for this.

File: examples/async/src/example/handler.clj [hunk 1]
```diff
@@ -55,4 +55,11 @@
                  (async/>! chan e))
                (finally
                  (async/close! chan)))))
-         chan)))))
+         chan)))
+   (context "/resource" []
+            (resource
```
Comment (Deraen) [inline examples/async/src/example/handler.clj]: Context should be indented with 2 spaces, like fn.

File: test/compojure/api/resource_test.clj [hunk 2]
```diff
@@ -134,6 +134,31 @@
...
+  @(handler  {:query-params {:x 1}}
+             #(reset! result %)
+             nil)
+  @(handler  {:query-params {:x -1}}
+             #(identity %)
+             #(reset! res-error %))
+  (handler  {:query-params {:x "x"}}
+            #(identity %)
+            #(reset! req-error %))
+
```
Comment (miikka) [inline test/compojure/api/resource_test.clj]: This looks a bit ugly, but the asynchronous computation has to be forced somehow. We discussed this with @kalekale and couldn't think of any obviously better solution, so let's have it like this.

File: examples/async/src/example/handler.clj [hunk 3]
```diff
@@ -45,7 +45,7 @@
      (GET "/divide" []
        :return {:result Float}
        :query-params [x :- Long, y :- Long]
-       :summary "multiply two numbers together"
+       :summary "divide two numbers together"
```
Comment (miikka) [inline examples/async/src/example/handler.clj]: Thanks for fixing this.

**Pull request merged**